addons/point_of_sale/models/pos_order.py

```diff
@@ -197,6 +197,18 @@ def _prepare_invoice_line(self, order_line):
             'product_uom_id': order_line.product_uom_id.id,
         }
 
+    def _prepare_invoice_lines(self):
```

Comment:
don't do that here. in function "_prepare_invoice_line" return multiple lines when needed.

And in the function _prepare_invoice_vals, get the list of invoice lines before creating the dict

Reply:
Can't put in `_prepare_invoice_line` as it return only one invoice line and changing the return type can be a problem if this is overriden elsewhere.